hydra-node/src/Hydra/Node.hs

```diff
@@ -197,7 +205,10 @@ stepHydraNode tracer node = do
   handleOutcome e = \case
     Error _ -> pure ()
     Wait _reason -> putEventAfter eq waitDelay (decreaseTTL e)
-    StateChanged sc -> append sc
+    StateChanged sc -> putEventToSinks eventSinks sc
+    -- not all events will produce a statechange
+    -- FIXME(Elaine): if we are going to use eventId instead of stateChangeId, here is where we'd change it
+    -- this FIXME noted only so that we consider this again at review time
```

Comment:
Should: try to use `eventId` also for `StateChanged` events

As discussed, I think the easiest would be to just use the `eventId` and we will automatically get a strictly, monotonically increasing identifier. Only caveat is that it would not be continuous.

If that is something we definitely want, I would suggest to add that counter "next to the HeadState" into the TVar wrapped by the `NodeState` handle. This would require another aggregate which gets loaded analogous to `recoverState` for the `HeadState` from a list of `StateChanged` (or `StateEvent` as I suggested above).

So maybe let's just start with the `eventId` and then change this further later?